### ISSUE TITLE
RuntimeError for main.py

### ISSUE THREAD

**Issue by pwy-cmd** [ChaoningZhang/Universal-Deep-Hiding#7]
Dear Chaoning:
    Thank you for your great job. I am struggling with an issue when I run the script `bash ./scripts/test_main.sh`. I put some images into the test folder and val folder in Dateset as cover and secret images. However, an issue appears. I don't know how to deal with the issus. Any help would be appreciated!
![sendpix4](https://user-images.githubusercontent.com/54989253/139077685-fd67edda-e414-4e41-8dd5-78f82b9e9754.jpg)

**Comment by NinoNeumann** [comment]
actually, just ignore the save_analysis_pic fuction, may be .... it will work.

**Comment by 751565516** [comment]
> 
Hello, ask for help. I have the same problem as the above. If you cancel this part, will it affect the accuracy of testing and training? Is your reproduction result OK. How to ignore this part?thx

**Comment by NinoNeumann** [comment]
> > 
> 
> Hello, ask for help. I have the same problem as the above. If you cancel this part, will it affect the accuracy of testing and training? Is your reproduction result OK. How to ignore this part?thx

I just looked at the code, and this function have no return. May be the function of `save_analysis_pic` is to save sth like the picture of the result.  So i don't think the ignorance will effect the accuracy, but lose the  pic of analysis  as a result.  Thanks for reply.

**Comment by pwy-cmd** [comment]
> > > 
> > 
> > 
> > Hello, ask for help. I have the same problem as the above. If you cancel this part, will it affect the accuracy of testing and training? Is your reproduction result OK. How to ignore this part?thx
> 
> I just looked at the code, and this function have no return. May be the function of `save_analysis_pic` is to save sth like the picture of the result. So i don't think the ignorance will effect the accuracy, but lose the pic of analysis as a result. Thanks for reply.

Yes, I agree. I found that the code is worked fine while I ignore these codes.

**Comment by zeabin** [comment]
My simple solution to this issue. It seems to save the picture correctly.
```
    # for i_cover in range(4):
    #     cover_i = cover[:,i_cover*opt.channel_cover:(i_cover+1)*opt.channel_cover,:,:]
    #     container_i = container[:,i_cover*opt.channel_cover:(i_cover+1)*opt.channel_cover,:,:]
    #     cover_gap_i = cover_gap[:,i_cover*opt.channel_cover:(i_cover+1)*opt.channel_cover,:,:]
    #     if i_cover == 0:
    #         showCover = torch.cat((cover_i, container_i, cover_gap_i),0)
    #     else:
    #         showCover = torch.cat((showCover, cover_i, container_i, cover_gap_i),0)
    
    # for i_secret in range(4):
    #     secret_i = secret[:,i_secret*opt.channel_secret:(i_secret+1)*opt.channel_secret,:,:]
    #     rev_secret_i = rev_secret[:,i_secret*opt.channel_secret:(i_secret+1)*opt.channel_secret,:,:]
    #     secret_gap_i = secret_gap[:,i_secret*opt.channel_secret:(i_secret+1)*opt.channel_secret,:,:]

    #     if i_secret == 0:
    #         showSecret = torch.cat((secret_i, rev_secret_i, secret_gap_i),0)
    #     else:
    #         showSecret = torch.cat((showSecret, secret_i, rev_secret_i, secret_gap_i),0)

    showCover = torch.cat((cover, container, cover_gap),0)
    showSecret = torch.cat((secret, rev_secret, secret_gap),0)
```
